src/transformers/quantizers/quantizer_bnb_8bit.py

```diff
@@ -210,6 +213,12 @@ def create_quantized_param(
             if unexpected_keys is not None:
                 unexpected_keys.remove(fp16_statistics_key)
 
+        if fp16_weights_format is not None:
+            # We just need to pop the `weight_format` keys from the state dict to remove unneeded
+            # messages. The correct format is correctly retrieved during the first forward pass.
+            if unexpected_keys is not None:
+                unexpected_keys.remove(fp16_weights_format_key)
```

Comment:
I think we can combine these two statements 

```suggestion
        # We just need to pop the `weight_format` keys from the state dict to remove unneeded
        # messages. The correct format is correctly retrieved during the first forward pass.
        if fp16_weights_format is not None and unexpected_keys is not None:
            unexpected_keys.remove(fp16_weights_format_key)
```